commit ef2986b34e4099bd48d3cc5590ab0f66fbf39f87
Author: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date:   Tue Jan 15 15:35:24 2019 +0000

    staging: bcm2835-camera: Add sanity checks for queue_setup/CREATE_BUFS
    
    Fixes a v4l2-compliance failure when passed a buffer that is
    too small.
    queue_setup wasn't handling the case where !(*nplanes), as
    used from CREATE_BUFS and requiring the driver to sanity
    check the provided buffer parameters. It was assuming that
    it was always being used in the REQBUFS case where it provides
    the buffer properties.
    
    Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>

diff --git a/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c b/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
index b8984dcfbfa7..dbdc4e84f3b1 100644
--- a/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
+++ b/drivers/staging/vc04_services/bcm2835-camera/bcm2835-camera.c
@@ -242,6 +242,22 @@ static int queue_setup(struct vb2_queue *vq,
 		return -EINVAL;
 	}
 
+	/* Handle CREATE_BUFS situation - *nplanes != 0 */
+	if (*nplanes) {
+		if (*nplanes != 1 ||
+		    sizes[0] < dev->capture.port->current_buffer.size) {
+			v4l2_dbg(1, bcm2835_v4l2_debug, &dev->v4l2_dev,
+				 "%s: dev:%p Invalid buffer request from CREATE_BUFS, size %u < %u, nplanes %u != 1\n",
+				 __func__, dev, sizes[0],
+				 dev->capture.port->current_buffer.size,
+				 *nplanes);
+			return -EINVAL;
+		} else {
+			return 0;
+		}
+	}
+
+	/* Handle REQBUFS situation */
 	size = dev->capture.port->current_buffer.size;
 	if (size == 0) {
 		v4l2_err(&dev->v4l2_dev,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1319_ef2986b34e4099bd48d3cc5590ab0f66fbf39f87.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1319_ef2986b34e4099bd48d3cc5590ab0f66fbf39f87.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
